docs/quickstarts/full-stack/nextjs.mdx

```diff
@@ -147,176 +185,63 @@ export default function Page() {
   By default, the Clerk components inherit the font family. With create-next-app they don't apply a global font family. When you launch your application you may notice that the font family is different.
 </Callout>
 
-### Update your dashboard settings
-
-To make sure you are using the embedded components you will need to update your instance settings in the [Clerk Dashboard](https://dashboard.clerk.dev) to use the embedded components.
 
-<Images
-  src="/images/common/path-setting.png"
-  width={1344}
-  height={523}
-  alt="Path setting Imagee"
-/>
+### Update your environment variables 
 
-## Add Middleware
+Next, add environment variables for the `signIn`, `signUp`, `afterSignUp` and `afterSignIn` paths:
 
-To use server side functionality, you need to add the Clerk middleware to your Next.js application. This allows you to access the session in your Next.js API routes and server-side rendered pages.
-
-```tsx filename="middleware.ts" copy
-import { withClerkMiddleware } from "@clerk/nextjs/server";
-import { NextResponse } from "next/server";
-import type { NextRequest } from "next/server";
+```sh copy filename=".env.local"
+NEXT_PUBLIC_CLERK_SIGN_IN_URL=/sign-in
+NEXT_PUBLIC_CLERK_SIGN_UP_URL=/sign-up
+NEXT_PUBLIC_CLERK_AFTER_SIGN_IN_URL=/
+NEXT_PUBLIC_CLERK_AFTER_SIGN_UP_URL=/
+    ```
 
-export default withClerkMiddleware((req: NextRequest) => {
-  return NextResponse.next();
-});
+These values control the behavior of the components when you sign in or sign up and when you click on the respective links at the bottom of each component.
 
-export const config = {
-  matcher: [
-    /*
-     * Match all request paths except for the ones starting with:
-     * - _next
-     * - static (static files)
-     * - favicon.ico (favicon file)
-     * - public folder
-     */
-    "/((?!static|.*\\..*|_next|favicon.ico).*)",
-    "/",
-  ],
-}
-```
+## Embed the <UserButton /> 
 
-## Protecting Your Pages
+The <UserButton /> allows you to manage your account information and log out, thus allowing you to complete a full authentication circle.
 
-Clerk offers two ways to protect your Next.js application, you can use Next.js Middleware or using our Control Components on the client side.
+You can add it anywhere you want, next to the logo in app/page.tsx is a good start.

```

Comment:
```suggestion
You can add it anywhere you want, but next to the logo in `app/page.tsx` is a good start.

```

Reply:
Note that we also support the `pages` directory here too, so we might need to call out both places.